styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -305,14 +305,7 @@ private void tryUpdatingCounter(RunState runState,
         .collect(toList());
 
     if (!depletedResourceIds.isEmpty()) {
-      final List<Resource> depletedResources = depletedResourceIds.stream().map(x ->
-          Resource.create(x, shardedCounter.getCounterSnapshot(x).getLimit()))
-          .collect(toList());
-      final Message message = Message.info(
-          String.format("Resource limit reached for: %s", depletedResources));
-      if (!runState.data().message().map(message::equals).orElse(false)) {
-        receiveIgnoreClosed(Event.info(runState.workflowInstance(), message), runState.counter());
-      }
+      MessageUtil.emitResourceLimitReachedMessage(this, runState, depletedResourceIds);
```

Comment:
This changes the content of the message from `Resource.toString()` to plain resource name, to align with `Scheduler`. Interesting, so we have been sending different messages for a while.

Reply:
Yeah, that's what prompted this refactor. To ensure that we send the same message. Is even in the PR description =)